rpcserver.go

```diff
@@ -1192,12 +1192,23 @@ func (r *rpcServer) EstimateFee(ctx context.Context,
 		return nil, err
 	}
 
+	coinSelectionStrategy, err := lnrpc.UnmarshallCoinSelectionStrategy(
+		in.CoinSelectionStrategy,
+		r.cfg.CoinSelectionStrategy,
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	// We will ask the wallet to create a tx using this fee rate. We set
 	// dryRun=true to avoid inflating the change addresses in the db.
 	var tx *txauthor.AuthoredTx
 	wallet := r.server.cc.Wallet
 	err = wallet.WithCoinSelectLock(func() error {
-		tx, err = wallet.CreateSimpleTx(outputs, feePerKw, minConfs, true)
+		tx, err = wallet.CreateSimpleTx(
+			outputs, feePerKw, minConfs,
+			coinSelectionStrategy, true,
```

Comment:
hmmm - i think this would have not compiled before this commit?

Reply:
Yes, that's true. Thanks for this! I'm pretty new to the multiple small commits approach and I wasn't aware that each commit must be compiled successfully I was only looking for the last commit to be compiled successfully. I've just made the following dependency graph diagram. This will help me with which changes to make first in the commit history also I'll keep in mind to run `./scripts/check-each-commit.sh upstream/master` command next time.
<img width="751" alt="Screenshot 2024-03-12 at 9 58 42 PM" src="https://github.com/lightningnetwork/lnd/assets/69568555/9c166c1b-65ed-47b8-89bf-927523d0be2a">